File: netlify/functions/services/reddit-client.js
/** Reddit API client for posting responses */
const snoowrap = require('snoowrap');
const {
  REDDIT_CLIENT_ID,
  REDDIT_CLIENT_SECRET,
  REDDIT_USERNAME,
  REDDIT_PASSWORD,
  REDDIT_USER_AGENT
} = require('./config');

class RedditClient {
  constructor() {
    if (!REDDIT_CLIENT_ID || !REDDIT_CLIENT_SECRET || !REDDIT_USERNAME || !REDDIT_PASSWORD) {
      throw new Error('Reddit API credentials are required');
    }

    this.reddit = new snoowrap({
      userAgent: REDDIT_USER_AGENT,
      clientId: REDDIT_CLIENT_ID,
      clientSecret: REDDIT_CLIENT_SECRET,
      username: REDDIT_USERNAME,
      password: REDDIT_PASSWORD
    });

    this._authVerified = false;
  }

  async _verifyAuth() {
    if (this._authVerified) {
      return;
    }
    try {
      await this.reddit.getMe();
      this._authVerified = true;
    } catch (error) {
      throw new Error(`Failed to authenticate with Reddit: ${error.message}`);
    }
  }

  async postComment(postId, commentText) {
    /** Post a comment on a Reddit post */
    await this._verifyAuth();
    try {
      const submission = this.reddit.getSubmission(postId);
      const comment = await submission.reply(commentText);
      return {
        success: true,
        comment_id: comment.id,
        comment_url: `https://reddit.com${comment.permalink}`
      };
    } catch (error) {
      console.error(`Error posting comment: ${error}`);
      return {
        success: false,
        error: error.message
      };
    }
  }

  async replyToComment(commentId, replyText) {
    /** Reply to a Reddit comment */
    await this._verifyAuth();
    try {
      const comment = this.reddit.getComment(commentId);
      const reply = await comment.reply(replyText);
      return {
        success: true,
        comment_id: reply.id,
        comment_url: `https://reddit.com${reply.permalink}`
      };
    } catch (error) {
      console.error(`Error replying to comment: ${error}`);
      return {
        success: false,
        error: error.message
      };
    }
  }

  async getPostContent(postId) {
    /** Get the full content of a Reddit post */
    await this._verifyAuth();
    try {
      const submission = this.reddit.getSubmission(postId);
      const submissionData = await submission.fetch();
      return {
        title: submissionData.title,
        selftext: submissionData.selftext,
        url: submissionData.url,
        permalink: `https://reddit.com${submissionData.permalink}`
      };
    } catch (error) {
      console.error(`Error fetching post: ${error}`);
      return null;
    }
  }
}

module.exports = RedditClient;

File: reddit_client.py
"""Reddit API client for posting responses"""
import praw
from config import (
    REDDIT_CLIENT_ID, REDDIT_CLIENT_SECRET, REDDIT_USERNAME,
    REDDIT_PASSWORD, REDDIT_USER_AGENT
)


class RedditClient:
    def __init__(self):
        if not all([REDDIT_CLIENT_ID, REDDIT_CLIENT_SECRET, REDDIT_USERNAME, REDDIT_PASSWORD]):
            raise ValueError("Reddit API credentials are required")
        
        self.reddit = praw.Reddit(
            client_id=REDDIT_CLIENT_ID,
            client_secret=REDDIT_CLIENT_SECRET,
            username=REDDIT_USERNAME,
            password=REDDIT_PASSWORD,
            user_agent=REDDIT_USER_AGENT
        )
        
        # Verify authentication
        try:
            self.reddit.user.me()
        except Exception as e:
            raise ValueError(f"Failed to authenticate with Reddit: {e}")
    
    def post_comment(self, post_id, comment_text):
        """Post a comment on a Reddit post"""
        try:
            submission = self.reddit.submission(id=post_id)
            comment = submission.reply(comment_text)
            return {
                'success': True,
                'comment_id': comment.id,
                'comment_url': f"https://reddit.com{comment.permalink}"
            }
        except Exception as e:
            print(f"Error posting comment: {e}")
            return {
                'success': False,
                'error': str(e)
            }
    
    def reply_to_comment(self, comment_id, reply_text):
        """Reply to a Reddit comment"""
        try:
            comment = self.reddit.comment(id=comment_id)
            reply = comment.reply(reply_text)
            return {
                'success': True,
                'comment_id': reply.id,
                'comment_url': f"https://reddit.com{reply.permalink}"
            }
        except Exception as e:
            print(f"Error replying to comment: {e}")
            return {
                'success': False,
                'error': str(e)
            }
    
    def get_post_content(self, post_id):
        """Get the full content of a Reddit post"""
        try:
            submission = self.reddit.submission(id=post_id)
            return {
                'title': submission.title,
                'selftext': submission.selftext,
                'url': submission.url,
                'permalink': f"https://reddit.com{submission.permalink}"
            }
        except Exception as e:
            print(f"Error fetching post: {e}")
            return None

